crates/holochain_p2p/tests/tests/blocks.rs

```diff
@@ -325,6 +326,120 @@ async fn block_is_scoped_per_dna() {
+        .unwrap();
+    actor_2
+        .join(dna_hash.clone(), agent_2.clone(), None)
+        .await
+        .unwrap();
+    actor_2.test_set_full_arcs(dna_hash.to_k2_space()).await;
+
+    // Exchange peer infos to accelerate bootstrapping.
+    tokio::time::timeout(Duration::from_secs(10), async {
+        loop {
+            let maybe_agent_info_1 = actor_1
+                .peer_store(dna_hash.clone())
+                .await
+                .unwrap()
+                .get(agent_1.to_k2_agent())
+                .await
+                .unwrap();
+            let maybe_agent_info_2 = actor_2
+                .peer_store(dna_hash.clone())
+                .await
+                .unwrap()
+                .get(agent_2.to_k2_agent())
+                .await
+                .unwrap();
+            if let (Some(agent_info_1), Some(agent_info_2)) =
+                (maybe_agent_info_1, maybe_agent_info_2)
+            {
+                actor_1
+                    .peer_store(dna_hash.clone())
+                    .await
+                    .unwrap()
+                    .insert(vec![agent_info_2])
+                    .await
+                    .unwrap();
+                actor_2
+                    .peer_store(dna_hash.clone())
+                    .await
+                    .unwrap()
+                    .insert(vec![agent_info_1])
+                    .await
+                    .unwrap();
+                break;
+            }
+            tokio::time::sleep(Duration::from_millis(100)).await;
+        }
+    })
+    .await
+    .unwrap();
+
+    let response = actor_1
+        .get(dna_hash.clone(), fixt!(ActionHash).into())
+        .await;
+    assert!(response.is_ok());
+
+    actor_1
+        .block(Block::new(
+            BlockTarget::Cell(
+                CellId::new(dna_hash.clone(), agent_2.clone()),
+                CellBlockReason::BadCrypto,
+            ),
+            InclusiveTimestampInterval::try_new(
+                holochain_timestamp::Timestamp::now(),
+                holochain_timestamp::Timestamp::max(),
+            )
+            .unwrap(),
+        ))
+        .await
+        .unwrap();
+
+    // Blocking removes an agent from the peer store.
+    // To make sure the block is being enforced, the agent should be re-added to the
+    // peer 1's peer store. That isn't possible, however, because the peer store
+    // implementation internally discards blocked agents when inserting.
+
+    // Attempt to insert the blocked agent into peer1's peer store anyway, even though
+    // it will not be successful.
+    let agent_info_2 = actor_2
+        .peer_store(dna_hash.clone())
+        .await
+        .unwrap()
+        .get(agent_2.to_k2_agent())
+        .await
+        .unwrap()
+        .expect("peer 2 must be present in peer store");
+
+    actor_1
+        .peer_store(dna_hash.clone())
+        .await
+        .unwrap()
+        .insert(vec![agent_info_2])
+        .await
+        .unwrap();
```

Comment:
I don't see the benefit of doing these steps?

Reply:
Okay, I can remove them.

Reply:
https://github.com/holochain/holochain/pull/5328/commits/92ba1e1ab895c8f75f676af8bff5e0d37b5ec468